control/subscription_group.go

```diff
@@ -383,7 +377,7 @@ func (s *subscriptionGroup) subscribePlugins(id string,
 	for i, sub := range plugins {
 		plg, err := s.pluginManager.get(key(sub))
 		if err != nil {
-			serrs = append(serrs, serror.New(err))
+			serrs = append(serrs, pluginNotFoundError(sub))
```

Comment:
Looking at the code, it's seems the check for a bad key is only if the plugin is not found and breaking the key apart to see if the latest plugin needs to be grabbed. The same error `bad key` is returned for multiple scenarios, so it's never clear what part of the key is bad. 

`pluginManager.get` could return a clearer error on what plugin was not found, so then creating this error in subscription_group.go would then not be needed.